Sources/WordPressKit/Services/ActivityServiceRemote.swift

```diff
@@ -152,18 +152,15 @@ open class ActivityServiceRemote: ServiceRemoteWordPressComREST {
                                         failure(ResponseError.decodingFailure)
                                     }
                                 }, failure: { error, _ in
-                                    // FIXME: A hack to support free WPCom sites and Rewind. Should be obsolote as soon as the backend
-                                    // stops returning 412's for those sites.
-                                    let nsError = error as NSError
-
-                                    guard nsError.domain == WordPressComRestApiEndpointError.errorDomain,
-                                       nsError.code == WordPressComRestApiErrorCode.preconditionFailure.rawValue else {
-                                        failure(error)
+                                    // FIXME: A hack to support free WPCom sites and Rewind.
+                                    // Should be obsolete as soon as the backend stops returning 412's for those sites.
+                                    guard let endpointError = error.castToEndpointErrorWitCode(.preconditionFailure) else {
+                                        success(RewindStatus(state: .unavailable))
                                         return
                                     }
 
-                                    let status = RewindStatus(state: .unavailable)
-                                    success(status)
+                                    failure(endpointError)
```

Comment:
The original code returns `error`. But there it's changed to `endpointError`. I don't think we should change it here?